firewall/inspection/portscan/detect.go

```diff
@@ -0,0 +1,357 @@
+package portscan
+
+import (
+	"context"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/safing/portbase/log"
+	"github.com/safing/portbase/modules"
+	"github.com/safing/portmaster/firewall/inspection"
+	"github.com/safing/portmaster/netenv"
+	"github.com/safing/portmaster/network"
+	"github.com/safing/portmaster/network/netutils"
+	"github.com/safing/portmaster/network/packet"
+	"github.com/safing/portmaster/process"
+	"github.com/safing/portmaster/status"
+)
+
+type tcpUDPport struct {
+	protocol packet.IPProtocol
+	port     uint16
+}
+
+type ipData struct {
+	score int //score needs to be big enough to keep maxScore + addScore... to prevent overflow
+	//	greylistingWorked bool
+	previousOffender bool
+	blocked          bool
+	ignore           bool
+	lastSeen         time.Time
+	lastUpdated      time.Time
+	blockedPorts     []tcpUDPport
+}
+
+const (
+	//fixme: Which production-values do we want to have?
+	cleanUpInterval = 1 * time.Minute //fixme: Debug-Value
+	cleanUpMaxDelay = 5 * time.Minute
+
+	startAfter            = 1 * time.Second //fixme: Debug Value; When should the Portscan Detection start to prevent blocking Apps that just try to reconnect?
+	decreaseInterval      = 11 * time.Second
+	unblockIdleTime       = 1 * time.Hour
+	undoSuspicionIdleTime = 24 * time.Hour
+	unignoreTime          = 24 * time.Hour
+
+	startRegisteredPorts = 1024
+	startDynamicPorts    = 32768
```

Comment:
Those variables sound like they start something. Consider renaming them `RegisteredPortRangeStart` and `DynamicPortRangeStart`. I guess the fact that windows uses a different port range for ephemeral ports isn't that important. Though, maybe we should add a comment.